CONTRIBUTING.md

```
@@ -0,0 +1,88 @@
+
+## Code Style
+
+- Run `cargo fmt` before committing.
+- Run `cargo clippy` and resolve all warnings.
+- Keep functions focused and small.
+
+## Commit Messages
+
+We recommend [Conventional Commits](https://www.conventionalcommits.org/):
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)